packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts

```diff
@@ -333,7 +333,7 @@ export class Trail extends Resource {
    * @default false
    */
   public logAllLambdaDataEvents(options: AddEventSelectorOptions = {}) {
-    return this.addEventSelector(DataResourceType.LAMBDA_FUNCTION, [ 'arn:aws:lambda' ], options);
+    return this.addEventSelector(DataResourceType.LAMBDA_FUNCTION, [ `arn:${Fn.ref('AWS::Partition')}:lambda` ], options);
```

Comment:
Use `Aws.PARTITION` instead of Fn.ref

Reply:
Done.

`Aws` is convenient for CFN pseudo parameters!